apps/api/src/hyp3_api/validation.py

```diff
@@ -70,31 +70,38 @@ def check_granules_exist(granules, granule_metadata):
         raise GranuleValidationError(f'Some requested scenes could not be found: {", ".join(not_found_granules)}')
 
 
-def check_dem_coverage(granule_metadata):
+def check_dem_coverage(_, granule_metadata):
     bad_granules = [g['name'] for g in granule_metadata if not has_sufficient_coverage(g['polygon'])]
     if bad_granules:
         raise GranuleValidationError(f'Some requested scenes do not have DEM coverage: {", ".join(bad_granules)}')
 
 
-def check_same_burst_ids(granule_metadata):
-    ref_burst_id, sec_burst_id = [granule['name'].split('_')[1] for granule in granule_metadata]
-    if ref_burst_id != sec_burst_id:
+def check_same_burst_ids(job, _):
+    ref_ids = [ref.split('_')[1] for ref in job['job_parameters'].get('reference', [])]
+    sec_ids = [sec.split('_')[1] for sec in job['job_parameters'].get('secondary', [])]
+    if ref_ids != sec_ids:
         raise GranuleValidationError(
-            f'The requested scenes do not have the same burst ID: {ref_burst_id} and {sec_burst_id}'
+            'The requested scenes have burst IDs with no matching pairs.'
+        )
```

Comment:
Do we think its worth extra code here to communicate which burst pairs don't match?

Reply:
I think that would be nice for the user, considering they may be submitting dozens of pairs. How about just loop through the pairs and report the first pair that doesn't match?

Reply:
Also, I find the wording of "The requested scenes have burst IDs with no matching pairs" confusing, but no need to revise it if it's going to be replaced by an error message for the specific pair that doesn't match.